### PR TITLE
fix(Input): add handling of input's ref

### LINKED ISSUE #1536: Can't get ref on <input> if it's a child of <Input>
Can't get React ref on `<input>` if it's a child of `<Input>`.

### Steps
```
<Input>
  <input type="text" ref={r => console.log(r)}/>
</Input>
```

### Expected Result
I would expect to see this following in the console, indicating that `r` is the ref to the `<input>` tag.
```
<input type="text">
```

### Actual Result
The ref function is never called. There is no console.log output at all.

### Version
0.67.2

### Testcase
http://codepen.io/darkadept/pen/mWazWa

### PR DESCRIPTION
**Pull request by layershifter**
Fixes #1536.

### CODE REVIEW

File: src/elements/Input/Input.js [hunk 1]
```diff
@@ -187,7 +192,10 @@ class Input extends Component {
       const childElements = _.map(Children.toArray(children), (child) => {
         if (child.type !== 'input') return child
 
-        return cloneElement(child, { ...htmlInputProps, ...child.props })
+        return createHTMLInput(child, {
+          defaultProps: htmlInputProps,
+          overrideProps: this.handleInputOverrides,
+        })
```
Comment (layershifter) [inline src/elements/Input/Input.js]: Use factory instead of `cloneElement`

File: test/specs/elements/Input/Input-test.js [hunk 2]
```diff
@@ -170,42 +186,39 @@ describe('Input', () => {
     })
   })
 
+  describe('ref', () => {
+    it('maintains ref on child node', () => {
+      const ref = sandbox.spy()
+      const wrapper = mount(<Input><input ref={ref} /></Input>)
+
+      // ref.should.have.been.calledOnce()
```
Comment (layershifter) [inline test/specs/elements/Input/Input-test.js]: @levithomason I can't get this test working, can you take a look?

Comment (levithomason) [reply test/specs/elements/Input/Input-test.js]: Refs are not passed through props. React makes them undefined. We'll have to accept the ref on a different prop name, or, update factories to include refs in the overrides. We should probably use a different prop name, such as, `inputRef`.

**Review by layershifter — COMMENTED** [review]
I've reverted back to `cloneElement` and updated handling of `ref`, also updated tests now everything works as expected. @levithomason thanks for the point to right direction

**Review by levithomason — APPROVED** [review]
Perfect, thanks!

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.68.0`.

**Pull request merged**